src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -193,4 +194,23 @@ export default Backbone.Model.extend({
 
     this.trigger('select:all');
   },
+  setLastSelectedAction(actionId) {
+    this.set('lastSelectedAction', actionId);
+  },
+  getLastSelected() {
+    return this.get('lastSelectedAction');
+  },
+  selectMultiple(arrayOfActionIds) {
+    const currentSelectedList = this.get('selectedActions');
+
+    const newSelectedList = arrayOfActionIds.reduce((selected, itemId) => {
+      selected[itemId] = true;
+      return selected;
+    }, currentSelectedList);
+
+    this.set('selectedActions', newSelectedList);
+
+    this.trigger('select:multiple');
+    this.trigger('change:selectedActions');
+  },
 });
```

Comment:
Takes the array of action `id`s and converts them into an object in the `{‘id’: true, ‘id’: true}` format we use in the state. Then sets that new list in the state.

Make sure to include the `id`s that already existed in the `selectedActions` object.

Some notes:

- I couldn’t get the state change event to trigger. So I’m manually doing `this.trigger(‘change:selectedActions’)` for that reason. Not sure why that isn’t triggering. Any ideas?

Reply:
Using `clone` seems to fix this.

```js
const newSelectedList = arrayOfActionIds.reduce((selected, itemId) => {
  selected[itemId] = true;
  return selected;
}, clone(currentSelectedList));
```

Or, using `Object.assign()` does as well:

```js
Object.assign({}, oldSelectedIds, newSelectedIds)
```

Reply:
ah right.. this was reducing the value already in the state